tools/RAiDER/models/hres.py

```diff
@@ -265,27 +224,116 @@ def _fetch(self, lats, lons, time, out, Nextra=2):
             self.update_a_b()
 
         # execute the search at ECMWF
-        self._download_ecmwf_file(lat_min, lat_max, self._lat_res, lon_min, lon_max, self._lon_res, time, out)
+        self._download_ecmwf(lat_min, lat_max, self._lat_res, lon_min, lon_max, self._lon_res, time, out)
 
-    def _download_ecmwf_file(self, lat_min, lat_max, lat_step, lon_min, lon_max, lon_step, time, out):
+    def _download_ecmwf(self, lat_min, lat_max, lat_step, lon_min, lon_max, lon_step, time, out):
         from ecmwfapi import ECMWFService
 
         server = ECMWFService("mars")
 
         corrected_date = round_date(time, datetime.timedelta(hours=6))
 
-        server.execute({
-            'class': self._classname,
-            'date': datetime.datetime.strftime(corrected_date, "%Y-%m-%d"),
-            'expver': "{}".format(self._expver),
-            'levelist': "1/to/{0}".format(self._levels),
-            'levtype': "ml",
-            'param': "129/130/133/152",
-            'stream': "oper",
-            'time': "{}".format(datetime.time.strftime(corrected_date.time(), '%H:%M')),
-            'type': "an",
-            'step': "0",
-            'grid': "{}/{}".format(lon_step, lat_step),
-            'area': "{}/{}/{}/{}".format(lat_max, lon_min, lat_min, lon_max),
-            'format': "netcdf", },
-            out)
+        if self._model_level_type == 'ml':
+            param = "129/130/133/152"
+        else:
+            param = "129.128/130.128/133.128/152"
+
+        server.execute(
+            {
+                'class': self._classname,
+                'dataset': self._dataset,
+                'expver': "{}".format(self._expver),
+                'resol': "av",
+                'stream': "oper",
+                'type': "an",
+                'levelist': "all",
+                'levtype': "{}".format(self._model_level_type),
+                'param': param,
+                'date': datetime.datetime.strftime(corrected_date, "%Y-%m-%d"),
+                'time': "{}".format(datetime.time.strftime(corrected_date.time(), '%H:%M')),
+                'step': "0",
+                'grid': "{}/{}".format(lon_step, lat_step),
+                'area': "{}/{}/{}/{}".format(lat_max, floorish(lon_min, 0.1), floorish(lat_min, 0.1), lon_max),
```

Comment:
If you floor should the max be ceiled?

Reply:
I don't think it's needed here; the only reason I floor is because I was getting API errors when allowing arbitrary floats. The maximum will already be a fixed integer multiple of the grid size, so ceil isn't needed as we already have a buffer.